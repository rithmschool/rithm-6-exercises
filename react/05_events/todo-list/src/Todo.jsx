import React, { Component } from "react";
import "./Todo.css";

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = { completed: false };
    this.markComplete = this.markComplete.bind(this);
    this.update = this.update.bind(this);
  }

  markComplete() {
    this.setState({ completed: !this.state.completed });
  }
  isEditing() {}
  update(e) {
    let newText = e.target.parentNodevalue;
    let statei = e.target.getAttribute("index");
    console.log(e.target.parentNode);
    console.log(newText, statei);
    // this.props.handleUpdate(newText, statei);
  }
  render() {
    var status = this.state.completed ? "line-through" : "none";
    return (
      <div className="Todo">
        <input onClick={this.markComplete} type="checkbox" />
        <p style={{ textDecoration: status }}>{this.props.text}</p>
        <input type="text" className="" />
        <button onClick={this.update}>Update</button>
        <button onClick={this.props.handleDelete}>Delete</button>
      </div>
    );
  }
}

export default Todo;
