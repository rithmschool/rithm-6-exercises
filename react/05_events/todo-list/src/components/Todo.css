.Todo {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 30rem;
}

.hidden {
  display: hidden;
}

p {
  flex-grow: 2;
}
