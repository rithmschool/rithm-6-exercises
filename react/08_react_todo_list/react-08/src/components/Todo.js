import React, { Component } from 'react';

class Todo extends Component {
  render() {
    return 'lol';
  }
}

export default Todo;
