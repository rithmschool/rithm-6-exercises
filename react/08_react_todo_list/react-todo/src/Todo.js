import React from 'react';

const Todo = ({
  title,
  description,
  isCompleted,
  markAsComplete,
  removeToDo,
  showEditForm,
  editToDo,
  isEditSelected
}) => (
  <div>
    <p>Title: {title}</p>
    <p>Description: {description}</p>

    <p>Is Completed: {isCompleted ? 'True' : 'False'}</p>
    <button onClick={markAsComplete}>Toggle Completion Status</button>
    <button onClick={removeToDo}>Delete</button>
    <button onClick={showEditForm}>Change</button>
    {isEditSelected ? (
      <form onSubmit={editToDo}>
        <label htmlFor="title">Title: </label>
        <input name="title" />
        <label htmlFor="description">Description: </label>
        <input name="description" />
        <input type="submit" />
      </form>
    ) : null}
  </div>
);

export default Todo;
