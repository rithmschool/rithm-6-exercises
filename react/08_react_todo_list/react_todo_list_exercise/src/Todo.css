.todo {
  width: 250px;
  margin: 0 auto;
}

.complete {
  text-decoration: line-through;
}
