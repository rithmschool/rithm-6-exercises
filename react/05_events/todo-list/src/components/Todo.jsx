import React, { Component } from "react";
import "./Todo.css";

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      completed: false,
      isEditing: false
    };
    this.markComplete = this.markComplete.bind(this);
    this.isEditing = this.isEditing.bind(this);
  }

  markComplete() {
    this.setState({ completed: !this.state.completed });
  }
  isEditing(e) {
    e.preventDefault();
    if (this.state.isEditing) {
      let newText = e.target.parentNode.children[2].children[0].value;
      if (newText !== "") {
        let statei = this.props.index;
        this.props.handleUpdate(newText, statei);
      }
    }
    this.setState({ isEditing: !this.state.isEditing });
  }
  render() {
    var status = this.state.completed ? "line-through" : "none";
    var edit = this.state.isEditing ? <input type="text" /> : <br />;
    return (
      <div className="Todo">
        <input onClick={this.markComplete} type="checkbox" />
        <p style={{ textDecoration: status }}>{this.props.text}</p>
        <form action="" onSubmit={this.isEditing}>
          {edit}
        </form>
        <button onClick={this.isEditing}>Update</button>
        <button onClick={this.props.handleDelete}>Delete</button>
      </div>
    );
  }
}

export default Todo;
