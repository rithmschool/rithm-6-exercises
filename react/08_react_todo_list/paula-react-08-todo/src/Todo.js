import React, { Component } from "react";
import "./Todo.css";

export default class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: this.props.title,
      description: this.props.description,
      editTitle: false,
      editDescription: false
    };
    this.onKeyDown = this.onKeyDown.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleDoubleClick(editKey) {
    let newState = this.state;
    newState[editKey] = true;
    this.setState({ newState });
  }

  onKeyDown(editKey, e) {
    if (e.key === "Enter") {
      this.setState({ [editKey]: false });
      this.handleSubmit(e);
    }
  }

  handleSubmit(e) {
    let updated = { field: e.target.name, value: e.target.value };
    this.props.handleEdit(updated);
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    return { title: nextProps.title, description: nextProps.description };
  }

  render() {
    const {
      title,
      description,
      isComplete,
      toggleComplete,
      handleDelete
    } = this.props;

    const strikeStyle = isComplete ? "line-through" : "none";

    const titleInput = (
      <input
        name="title"
        value={this.state.title}
        onChange={this.handleChange}
        onKeyDown={this.onKeyDown.bind(this, "editTitle")}
      />
    );
    const descInput = (
      <input
        name="description"
        value={this.state.title}
        onChange={this.handleChange}
        onKeyDown={this.onKeyDown.bind(this, "editDescription")}
      />
    );
    const titleSpan = (
      <span onDoubleClick={this.handleDoubleClick.bind(this, "editTitle")}>
        {title}:{" "}
      </span>
    );
    const descSpan = (
      <span
        onDoubleClick={this.handleDoubleClick.bind(this, "editDescription")}
      >
        {description}
      </span>
    );
    const titleDisplay = this.state.editTitle ? titleInput : titleSpan;
    const descDisplay = this.state.editDescription ? descInput : descSpan;

    return (
      <div>
        <h6 style={{ textDecoration: strikeStyle }}>
          {titleDisplay} {descDisplay}
        </h6>
        <button onClick={toggleComplete}>Mark as Complete!</button>
        <button onClick={handleDelete}>Delete Me!</button>
      </div>
    );
  }
}
