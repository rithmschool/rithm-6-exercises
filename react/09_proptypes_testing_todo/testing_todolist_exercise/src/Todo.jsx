import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Todo.css"

export default class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: "",
      description: "",
    }
    this.handleEditSubmit = this.handleEditSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    return {
      title: nextProps.title,
      description: nextProps.description
    }
  }

  handleEditSubmit(e) {
    e.preventDefault();
    this.props.handleEdit(this.state);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  render() {
    return (
      !this.props.isEditing ? (
        <li className={`todo ${this.props.className}`} onDoubleClick={this.props.handleEditRequest}>
          <p>Title: {this.props.title}</p>
          <p>Description: {this.props.description}</p>
          <button onClick={this.props.handleComplete}>Mark/Unmark as Complete</button>
          <button onClick={this.props.handleDelete}>Delete Item</button>
        </li>
      ) : (
          <form onSubmit={this.handleEditSubmit}>
            <li className={`todo ${this.props.className}`}>

              <label htmlFor="title">Title: </label>
              <input name="title" value={this.state.title} onChange={this.handleChange} />

              <label htmlFor="description">Description: </label>
              <input name="description" value={this.state.description} onChange={this.handleChange} />

              <button>Submit</button>

            </li>
          </form>
        )
    );
  }
}

Todo.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  className: PropTypes.string,
  isEditing: PropTypes.string,
  handleEdit: PropTypes.func,
  handleDelete: PropTypes.func,
  handleComplete: PropTypes.func,
  handleEditRequest: PropTypes.func
};
