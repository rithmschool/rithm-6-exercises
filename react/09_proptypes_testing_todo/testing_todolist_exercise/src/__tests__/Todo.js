import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Todo from '../Todo';

describe('<Todo />', () => {

  let wrapper;

  it('renders', () => {
    wrapper = mount(<Todo />);
  });

  it('matches snapshot', () => {
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  })

})
