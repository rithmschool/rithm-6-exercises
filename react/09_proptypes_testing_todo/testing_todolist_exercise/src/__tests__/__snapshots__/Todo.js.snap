// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> matches snapshot 1`] = `
<Todo>
  <li
    className="todo undefined"
  >
    <p>
      Title: 
    </p>
    <p>
      Description: 
    </p>
    <button>
      Mark/Unmark as Complete
    </button>
    <button>
      Delete Item
    </button>
  </li>
</Todo>
`;
